jscomp/others/js_option.ml

```diff
@@ -59,5 +59,27 @@ let andThen f x =
   | None -> None 
   | Some x -> f x [@bs]
 
+let map f x =
+  match x with
+  | None -> None
+  | Some x -> Some (f x [@bs])
```

Comment:
Plus this would make a good usecase for filter too!  ^.^

```ocaml
let is_valid_month month =
  month >= 1 && month <= 12

let to_valid_month month =
  month |> Option.filter is_valid_month

let head = function
  | [] -> None
  | first :: _ -> Some first

let get_first_month months =
  head months
  |> to_valid_month
```
So now `to_valid_month` is not doing the dual-task of both testing a predicate and packing a value (thus the predicate can be used elsewhere now), and piping into functions straight is always better when possible.

This is a side note though, but still, `flat_map` is still more descriptive than `and_then`, which just makes no grammatical sense, plus is an actual standard word.  ;-)

Could just do both though.  :-)